src/Output/StrictUnifiedDiffOutputBuilder.php

```diff
@@ -0,0 +1,319 @@
+                    if ("\n" !== $lc) {
+                        \array_splice($diff, $i + 1, 0, [["\n\\ No newline at end of file\n", self::$noNewlineAtOEFid]]);
+                    }
+
+                    if (!\count($toFind)) {
+                        break;
+                    }
+                }
+            }
+        }
+
+        // write hunks to output buffer
+
+        $cutOff      = \max($this->commonLineThreshold, $this->contextLines);
+        $hunkCapture = false;
+        $sameCount   = $toRange = $fromRange = 0;
+        $toStart     = $fromStart = 1;
+
+        foreach ($diff as $i => $entry) {
+            if (0 === $entry[1]) { // same
+                if (false === $hunkCapture) {
+                    ++$fromStart;
+                    ++$toStart;
+
+                    continue;
+                }
+
+                ++$sameCount;
+                ++$toRange;
+                ++$fromRange;
+
+                if ($sameCount === $cutOff) {
+                    $contextStartOffset = ($hunkCapture - $this->contextLines) < 0
+                        ? $hunkCapture
+                        : $this->contextLines
+                    ;
+
+                    // note: $contextEndOffset = $this->contextLines;
+                    //
+                    // because we never go beyond the end of the diff.
+                    // with the cutoff/contextlines here the follow is never true;
+                    //
+                    // if ($i - $cutOff + $this->contextLines + 1 > \count($diff)) {
+                    //    $contextEndOffset = count($diff) - 1;
+                    // }
+                    //
+                    // ; that would be true for a trailing incomplete hunk case which is dealt with after this loop
+
+                    $this->writeHunk(
+                        $diff,
+                        $hunkCapture - $contextStartOffset,
+                        $i - $cutOff + $this->contextLines + 1,
+                        $fromStart - $contextStartOffset,
+                        $fromRange - $cutOff + $contextStartOffset + $this->contextLines,
+                        $toStart - $contextStartOffset,
+                        $toRange - $cutOff + $contextStartOffset + $this->contextLines,
+                        $output
+                    );
+
+                    $fromStart += $fromRange;
+                    $toStart += $toRange;
+
+                    $hunkCapture = false;
+                    $sameCount   = $toRange = $fromRange = 0;
+                }
+
+                continue;
+            }
+
+            $sameCount = 0;
+
+            if ($entry[1] === self::$noNewlineAtOEFid) {
+                continue;
+            }
+
+            $this->changed = true;
+
+            if (false === $hunkCapture) {
+                $hunkCapture = $i;
+            }
+
+            if (1 === $entry[1]) { // added
+                ++$toRange;
+            }
+
+            if (2 === $entry[1]) { // removed
```

Comment:
next reason to consider using constants from some enum instead of magic numbers

Reply:
agreed, however I follow the preference here:
https://github.com/sebastianbergmann/diff/pull/2

Reply:
when it was used only in one class, it was more obvious
now, when it's used across different classes, It's worth to have them declared in one place.
less docs (like here `// removed`), less chances to forgot to change one of the values, easier to grep for.
Also, I do agree that `static::SOME_CONST` could be problematic in performance due to LSB, yet with enum approach, they will be always accessible like `MyEnum::SOME_CONST`, no